File: ApplicationAlgorithm/test/unit3/unit1.cpp
#include <iostream>

using namespace std;

// Phân tích số n thành tổng của đúng m số nguyên dương
void function(int n, int m)
{
    if (n < m)
        return;
    else
    {
        int a[m];
        int j = 1;
        for (int i = 0; i < n; i++)
        {
            a[i] = j;
            a[i + 1] = m - j;
        }
        while (n - m >= m - 1)
        {
            cout << "";
            for (int i = 0; i < n; i++)
            {
                cout << a[i] << "+";
            }
        }
        
    }
}

int main()
{
    int n, m;
    cout << "Nhap n = ";
    cin >> n;
    cout << "Nhap m = ";
    cin >> m;
    function(n, m);
}

File: ApplicationAlgorithm/src/py/unit1.py
# Nhập 2 số nguyên a,b
# Tính luỹ thừa b của a (a^b)
# Nếu giá trị tìm được quá lớn in ra 9 chữ số cuối cùng
def function(a, b):
    e = pow(a, b)
    if e < 1000000000:
        return e
    else:
        return str(e)[-9:] 

a = int(input("Nhap a = "))
b = int(input("Nhap b = "))
print(function(a, b))